File: sukh_1684/problems/longest_common_prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        n = len(strs)
        for i in range(1,n):
            s = min(len(prefix),len(strs[i]))
            prefix = prefix[:s]
            for j in range(s):
                if (strs[i][j] != prefix[j]):
                    prefix = prefix[:j]
                    break
                    
        return prefix

File: sukh_1684/problems/longest_common_prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==0){
            return "";
        }
        String s=strs[0];
          int i;
      
        for( i=0;i<s.length();i++){
            char c=s.charAt(i);

            for(int j=1;j<strs.length;j++){
                if(strs[j].length()==0){
                    return "";
                }
                if(strs[j].length()==i){
                    return s.substring(0,i);
                }
                if(c!=strs[j].charAt(i)){
                    return s.substring(0,i);
                }
            }
        }
                if(i==s.length()){
                        return s;
                    }
        
      return "" ;
    }
}